### PR TITLE
[No Review Needed] Add publishing infra version 3

### PR DESCRIPTION
**Pull request by JohnTortugo**
This other PR needs to be merged first: https://github.com/dotnet/arcade/pull/5711

This needs to pass these tests before merging:

- [x] Publish a build using V2 infra:
- [x] Publish a build using V3 infra:
- [x] Promote a build using V3 infra:

### CODE REVIEW

File: Arcade.sln [hunk 1]
```diff
@@ -5,100 +5,12 @@ VisualStudioVersion = 16.0.29019.234
 MinimumVisualStudioVersion = 10.0.40219.1
```
Comment (JohnTortugo) [inline Arcade.sln:5]: Need to revert this.

File: NuGet.config [hunk 2]
```diff
@@ -8,6 +8,7 @@
     <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
     <add key="dotnet-tools" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json" />
     <add key="dotnet-eng" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json" />
+    <add key="general-testing" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/general-testing/nuget/v3/index.json" />
```
Comment (JohnTortugo) [inline NuGet.config:11]: Need to revert this.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -199,15 +199,12 @@ stages:
...
     parameters:
-      # Symbol validation isn't being very reliable lately. This should be enabled back
-      # once this issue is resolved: https://github.com/dotnet/arcade/issues/2871
-      enableSymbolValidation: false
+      publishingInfraVersion: 2
+
       # Sourcelink validation isn't passing for Arcade due to some regressions. This should be
       # enabled back once this issue is resolved: https://github.com/dotnet/arcade/issues/2912
       enableSourceLinkValidation: false
-      publishDependsOn:
```
Comment (JohnTortugo) [inline azure-pipelines.yml:208]: Need to revert this.

File: azure-pipelines.yml [hunk 4]
```diff
@@ -199,15 +199,12 @@ stages:
 
   - template: eng\common\templates\post-build\post-build.yml
     parameters:
-      # Symbol validation isn't being very reliable lately. This should be enabled back
-      # once this issue is resolved: https://github.com/dotnet/arcade/issues/2871
-      enableSymbolValidation: false
+      publishingInfraVersion: 2
```
Comment (JohnTortugo) [inline azure-pipelines.yml:202]: Need to revert this.

File: eng/Publishing.props [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<Project>
```
Comment (JohnTortugo) [inline eng/Publishing.props:1]: Need to revert this.

File: eng/common/templates/post-build/post-build.yml [hunk 6]
```diff
@@ -81,6 +91,7 @@ stages:
           inputs:
             filePath: $(Build.SourcesDirectory)/eng/common/post-build/publish-using-darc.ps1
             arguments: -BuildId $(BARBuildId) 
+              -PublishingInfraVersion ${{ parameters.PublishingInfraVersion }}
               -AzdoToken '$(publishing-dnceng-devdiv-code-r-build-re)'
```
Comment (JohnTortugo) [inline eng/common/templates/post-build/post-build.yml:95]: @epananth - I think this secret expired and it caused the error in my last validation build.

Comment (epananth) [reply eng/common/templates/post-build/post-build.yml:95]: this secret is recycled.

**Pull request closed without merging**